enhancements/storage/csi-driver-global-configmap-secret-mounter.md

```diff
@@ -0,0 +1,201 @@
+---
+title: global-configmap-secret-mounter
+authors:
+  - "@janedoe"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Global ConfigMap and Secret Mounter
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+ 1.  It is possible to build some sorts of per-namespace ACL boundaries if we really wanted to, but I think it makes more sense
+ to have the idea of global or namespaced and no in-between since nothing else is in between today.  They can write a controller
+ if they really want to create resources shared across some namespaces and not others.
+
+## Summary
+
+A CSI driver that can mount a configmap or secret from a specified namespace into any pod.
+This would allow a single copy of a configmap to exist in openshift-config-managed (say the trusted-ca) and mount that configmap into any pod.
+There is no ACL check on the pod attempting to mount it.  Any configmap or secret that can be mounted, can be mounted anywhere.
+The CSI driver must keep the content on disk up to date, similar to how the kubelet keeps configmap and secret data consistent today.
+This allows the creation of a webhook admission plugin to allow annotation based injection of well-known configmaps and secrets, like the trusted-ca today.
+
+To have reasonable security boundaries, the cluster-admin must explicitly enumerate all namespace/name tuples for the configmaps and secrets that he wants to expose. 
+This will allow the CSI driver to use individual list/watches to drive the watch behavior to have scoped RBAC for the CSI driver.
+
+## Motivation
+
+This CSI driver would generically solve the problem of "inject this custom CA bundle into every requesting pod".
+A clever admission webhook could even do this unconditionally across namespaces or even have a bash wrapping ability for system certs if so desired. 
```

Comment:
barring the creation of the admission webhook (i think you indicated that could/would be later work), how does someone consume this?  just explicitly add the right volume/mount to their pod?

Where will the decision about the filestructure of the content being mounted be made?  Is that something the CSI driver does, or it's entirely controlled by the pod's volume mount spec?  (Thinking about cases where just directly mounting the configmap as is will not be sufficient for some consumers, especially depending on the structure of the CA configmap that we settle on).

(presumably if it's determined by the volume mount spec, then the admission webhook could also drive it, which mans the consumer can potentially drive it by well-crafted annotations that drive the webhook?  but i'd definitely like to better understand the end-to-end consumption experience)